node/network/src/peer_manager.rs

```diff
@@ -124,13 +125,14 @@ impl<T: ChainStateRetriever> PeerManager<T> {
         tokio::spawn(task);
 
         // Spawn the task that listens to incoming connections.
+        let node_info1 = node_info.clone();
         let all_peer_states3 = all_peer_states.clone();
         let task = TcpListener::bind(&node_info.addr)
             .expect("Cannot listen to the address")
             .incoming()
             .for_each(move |socket| {
                 Peer::spawn_incoming_conn(
-                    node_info.clone(),
+                    node_info1.clone(),
```

Comment:
cloning second time?

Reply:
Yes, because it it consumed by `Peer::spawn_incoming_conn`. This code does N+1 clones where N is the number of iterations of `for_each`. It is slightly excessive, and can be fixed by `fold`, but I do not really want to be doing it now.